sim/rogue/main_gauche.go

```diff
@@ -37,9 +37,31 @@ func (rogue *Rogue) registerMainGaucheSpell() {
 		Duration: time.Second * 10,
 		OnGain: func(aura *core.Aura, sim *core.Simulation) {
 			rogue.SinisterStrike.Cost.FlatModifier -= 20
+			rogue.SinisterStrike.ThreatMultiplier *= 1.5
+
+			if rogue.HasRune(proto.RogueRune_RunePoisonedKnife) {
```

Comment:
`HasRune` checks can be expensive when done in requently-called callbacks. Could you store both rune checks in variables above then use the variables instead?

Reply:
cleaned up